Userland/Libraries/LibPDF/Filter.cpp

```diff
@@ -56,29 +56,33 @@ PDFErrorOr<ByteBuffer> Filter::decode(ReadonlyBytes bytes, DeprecatedFlyString c
 
 PDFErrorOr<ByteBuffer> Filter::decode_ascii_hex(ReadonlyBytes bytes)
 {
-    if (bytes.size() % 2 == 0)
-        return TRY(decode_hex(bytes));
-
-    // FIXME: Integrate this padding into AK/Hex?
-
-    auto output = TRY(ByteBuffer::create_zeroed(bytes.size() / 2 + 1));
-
-    for (size_t i = 0; i < bytes.size() / 2; ++i) {
-        auto const c1 = decode_hex_digit(static_cast<char>(bytes[i * 2]));
-        if (c1 >= 16)
-            return AK::Error::from_string_literal("Hex string contains invalid digit");
-
-        auto const c2 = decode_hex_digit(static_cast<char>(bytes[i * 2 + 1]));
-        if (c2 >= 16)
-            return AK::Error::from_string_literal("Hex string contains invalid digit");
-
-        output[i] = (c1 << 4) + c2;
+    ByteBuffer output;
+
+    bool have_read_high_nibble = false;
```

Comment:
how do you decide youʼre starting with high nibble? shouldnʼt you check parity of bytes length to accurately decide that?

Reply:
The spec guarantees that. The only special case is that if you encountered an odd number of digits, you "append" a zero.
Can't use the byte length at all, because we can have any amount of whitespace in between the digits.

Reply:
neat; thanks for sharing details! 🙇🏻‍♂️